test/manipulation.test.js

```diff
@@ -544,31 +544,72 @@ describe('manipulation', function() {
         });
     });
 
-    it('should ignore unknown attributes when strict: true', function(done) {
-      // Using {foo: 'bar'} only causes dependent test failures due to the
-      // stripping of object properties when in strict mode (ie. {foo: 'bar'}
-      // changes to '{}' and breaks other tests
-      person.updateAttributes({name: 'John', foo: 'bar'},
-        function(err, p) {
-          if (err) return done(err);
-          should.not.exist(p.foo);
-          Person.findById(p.id, function(e, p) {
-            if (e) return done(e);
-            should.not.exist(p.foo);
-            done();
-          });
+    it('should discard undefined values before strict validation',
+      function(done) {
+        Person.definition.settings.strict = true;
+        Person.findById(person.id, function(err, p) {
+          p.updateAttributes({name: 'John', unknownVar: undefined},
+            function(err, p) {
+              // if uknownVar was defined, it would return validationError
+              if (err) return done(err);
+              Person.findById(p.id, function(e, p) {
+                if (e) return done(e);
+                p.name.should.equal('John');
+                should.not.exist(p.unknownVar);
+                done();
+              });
+            });
         });
-    });
+      });
 
-    it('should throw error on unknown attributes when strict: throw', function(done) {
+    // `strict: true` used to silently remove unknown properties,
+    // now return validationError upon unknown properties,
+    // same as previous validate behavior
```

Comment:
This comment does not make sense to me when it's before `it should allow unknown attributes when strict: false`